NetworkLayer/Model/CoreCategoryService.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  CoreCategoryService.swift
+//  NetworkLayer
+//
+//  Created by Clément Garcia on 17/10/2022.
+//
+
+import Foundation
+
+final class CoreCategoryService {
+    
+    // MARK: - Vars
+    /// URLSession handling the call to the REST apis
+    private let session: URLSession
+    
+    // MARK: - initializer
+    init(session: URLSession = .shared) {
+        self.session = session
+    }
+    
+    /// Retreive the categories list
+    /// - Returns: List of category
+    func getCategoryList() async throws -> CategoryList {
+        let (data, _) = try await session.data(from: ApiEndpoint.getEndpointCategory())
```

Comment:
```suggestion
        let (data, response) = try await session.data(from: ApiEndpoint.getEndpointCategory())
```

Même chose ici, si tu veux être complet tu dois vérifier la réponse.